javascript/javascript1/week3/main.js

```diff
@@ -0,0 +1,272 @@
+    }
+    const totalPercent = ((minsSpent/minsInLifespan) * 100).toFixed(3);
+    console.log(`In total that is ${totalPercent}% of my life`);
+  }
+  
+logOutSeriesText(seriesDurations);
+
+//NOnoN0nOYes (Note taking app)
+
+//Save a note
+const notes = [];
+
+function saveNote(content, id) {
+  // write some code here
+  return notes.push({content, id});
+}
+
+saveNote("Pick up groceries", 1);
+saveNote("Do laundry", 2);
+
+console.log(notes); // [{content: 'Pick up groceries', id: 1}, {content: 'Do laundry', id: 2}]
+
+//Get a note
+function getNote(id) {
+  // your code here
+  if (id === undefined || isNaN(id)) {
+    return console.error("invalid ID or ID not found");
+  }
+
+  for (let i = 0; i < notes.length; i++) {
+    const note = notes[i];
+    if (note.id === id) {
+    return note;
+    }
+  }
+  return console.error(`Note with ID ${id} not found`);
+
+}
+  const firstNote = getNote(1);
+  console.log(firstNote); // {content: 'Pick up groceries', id: 1}
+  
+//Log out notes
+function logOutNotesFormatted() {
+    // your code here
+    for (i = 0; i < notes.length; i++) {
+    const note = notes[i];
+    console.log(`The note with id ${note.id}, has the following note text: ${note.content}`);
+    }
+    return notes;
+}
+logOutNotesFormatted(); // should log out the text below
+
+// The note with id: 1, has the following note text: Pick up groceries
+// The note with id: 2, has the following note text: Do laundry
+
+// Unique feature 
+function jobIsDone(id) { // show note that is done and show notes that remained
+  if (id === undefined || isNaN(id)) {
+    return console.error("invalid ID or ID not found");
+  }
+
+  for (let i = 0; i < notes.length; i++) {
+    const note = notes[i];
+    if (note.id === id) {
+    notes.splice(i, 1);
+    console.log(`The note with id ${note.id} and content ${note.content} is done!`);
+    return logOutNotesFormatted();
+    }
+  }
+  return console.error(`Note with ID ${id} not found`);
+}
+
+jobIsDone(2);
+
+//CactusIO-interactive (Smart phone usage app)
+//Adding an activity
+const activities = [];
+const limitUsage = 180; // usage limit is 180 min
+
+function addActivity(date, activity, duration) {
+    let actObj = {
+        date,
+        activity,
+        duration,
+    };
+    activities.push(actObj);
+}
+showStatus(activities);
+addActivity("23/7-18", "Youtube", 30);
+
+//Show my status
+function showStatus (activities) {
+
+    if (activities.length === 0) {
+        console.error("Add some activities before calling showStatus");
+        return null;
+    }
+
+    let totalAmount = activities.length;
+    let totalMins = 0;
+
+    for (i = 0; i < activities.length; i++) {
+        totalMins += activities[i].duration;
+    }
+
+    console.log(`You have added ${totalAmount} activities. They amount to ${totalMins} min. of usage`);
+    //Usage limit
+    totalMins > limitUsage ? 
+    console.log("You have reached your limit, no more smartphoning for you!") :
+    "";
```

Comment:
As you are not returning anything there is no need for the ternary operator. You can just use if, or, if you are curious, you can do the same with && operator